twenty-first/src/shared_math/polynomial.rs

```diff
@@ -2754,4 +2804,86 @@ mod test_polynomials {
             assert_eq!(zerofier_polynomial, fast_zerofier_polynomial);
         }
     }
+
+    #[test]
+    fn differentiate_zero() {
+        let elm = BFieldElement::new(0);
+        let p = Polynomial::new_const(elm);
+        let q = p.formal_derivative();
+
+        assert!(q.is_zero());
+        assert_eq!(q.degree(), -1)
+    }
+    #[test]
+
```

Comment:
Clippy was happy with this? :fearful: 
```suggestion

    #[test]
```